test/upsample.jl

```diff
@@ -13,32 +13,65 @@
 end
 
 @testset "upsample_bilinear 2d" begin
-    x = reshape(Float32[1. 2.; 3. 4.], (2,2,1,1))
-    y_true =   [1//1  5//4    7//4    2//1;
-                1//1  5//4    7//4    2//1;
-                5//3  23//12  29//12  8//3;
-                7//3  31//12  37//12  10//3;
-                3//1  13//4   15//4   4//1;
-                3//1  13//4   15//4   4//1][:,:,:,:]
-   
+    x = Float32[1 2; 3 4][:,:,:,:]
+    x = cat(x,x; dims=3)
+    x = cat(x,x; dims=4)
+
+    # this output matches the one of pytorch v1.5.0
+    # nn.UpsamplingBilinear2d(scale_factor=(3,2), align_corners=True)
+    # for above x
+    y_true = Float32[ 1//1  4//3   5//3   2//1;
+                      7//5 26//15 31//15 12//5;
+                      9//5 32//15 37//15 14//5;
+                     11//5 38//15 43//15 16//5;
+                     13//5 44//15 49//15 18//5;
+                      3//1 10//3  11//3   4//1][:,:,:,:]
+    y_true = cat(y_true,y_true; dims=3)
+    y_true = cat(y_true,y_true; dims=4)
+
     y = upsample_bilinear(x, (3, 2))
     @test size(y) == size(y_true)
     @test eltype(y) == Float32
     @test y ≈ y_true
 
-    gradtest(x->upsample_bilinear(x, (3, 2)), x, atol=1e-4)
-
-    if CUDA.has_cuda()
-        y = upsample_bilinear(x |> cu, (3, 2))
-        @test y isa CuArray 
-        @test Array(y) ≈ y_true
-        g_gpu = Zygote.gradient(x -> sum(sin.(upsample_bilinear(x, (3, 2))))
-                                , x |> cu)[1]
-        @test g_gpu isa CuArray
-        g_cpu = Zygote.gradient(x -> sum(sin.(upsample_bilinear(x, (3, 2))))
-                                , x)[1]
-        @test Array(g_cpu) ≈ g_cpu  atol=1e-4
-    end
+    gradtest(x->upsample_bilinear(x, (3, 2)), x, atol=1e-3) # works to higher precision for Float64
+
+    # additional grad check, also compliant with pytorch
+    o = ones(Float32,6,4,2,1)
+    grad_true = 6*ones(Float32,2,2,2,1)
+    @test ∇upsample_bilinear(o, (3,2)) ≈ grad_true
```

Comment:
Step N.... I think this is the only test which needs changing? 
```suggestion
    @test ∇upsample_bilinear(o; size = (2,2)) ≈ grad_true
```